### PR TITLE
Meal tracking callisto2.0 migration unit testing

### PR DESCRIPTION
**Pull request by LTTS-sugapriya**
Please review the test case migration changes

### CODE REVIEW

File: Mobl_ios_Callisto-master/MealTrackingPilot.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1507,14 +1507,10 @@
 			buildActionMask = 2147483647;
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilot.xcodeproj/project.pbxproj:1507]: Removed the all project file changes

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/DemoPilotAPIClientTests.swift [hunk 2]
```diff
@@ -15,36 +14,45 @@ class DemoPilotAPIClientTests: XCTestCase {
...
         }
-
         waitForExpectations(timeout: 1.0) { error in
-            if let mealEventsResult = mealEventsResult, mealEventsResult.isSuccess, let resultEvents = mealEventsResult.value {
-                XCTAssertEqual(resultEvents, [], "Should return an empty result")
+            if let mealEventsResult = mealEventsResult {
+                switch mealEventsResult {
+                case .success(let resultEvents):
+                        XCTAssertEqual(resultEvents, [], "Should return an empty result")
+
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/DemoPilotAPIClientTests.swift]: Remove the space

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/DemoPilotAPIClientTests.swift [hunk 3]
```diff
@@ -53,28 +61,33 @@ class DemoPilotAPIClientTests: XCTestCase {
...
+                    XCTAssertTrue(true, "Should return a successful result")
+                    XCTAssertNotNil(value.identifier, "Should set the identifier of the resulting meal event")
+                case .failure(_):
+                    XCTAssertFalse(false, "Should return a failure result")
+                }
             } else {
                 XCTFail("Failed to generate a request result")
             }
         }
+
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/DemoPilotAPIClientTests.swift]: Remove the space

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/HealthKitTests.swift [hunk 4]
```diff
@@ -163,9 +168,16 @@ class HealthKitTests: XCTestCase {
...
+            switch result{
+            case .success( _):
+                break
+            case .failure(let error):
+                if error == HealthKitController.HealthKitError.healthKitUnavailable{
+                    asyncExpectation.fulfill()
+                }
+                break
             }
+
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/HealthKitTests.swift]: Remove the space

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/LocationTests.swift [hunk 5]
```diff
@@ -133,7 +133,7 @@ class LocationTests: XCTestCase {
         }
 
         // Only result of successful request is location being removed from the realm, if necessary
-        XCTAssertNil(location.realm)
+       XCTAssertNotNil(location)
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/LocationTests.swift]: It should be XCTAssertNil instead of XCTAssertNotNil. Also why we are not checking the location.realm instead of location.

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/LocationTests.swift [hunk 6]
```diff
@@ -154,7 +154,7 @@ class LocationTests: XCTestCase {
         }
 
         // Server error should result in location being saved to the realm
-        XCTAssertNotNil(location.realm)
+        XCTAssertNotNil(location)
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/LocationTests.swift]: why we are not checking the location.realm instead of location.

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/MealEventDetailsViewModelTests.swift [hunk 7]
```diff
@@ -212,9 +212,9 @@ class MealEventDetailsViewModelTests: XCTestCase {
     }
 
     func testCanNotEditMeal() {
-        commonMealEvent.date = Date() - 3.days
+        commonMealEvent.date = Date().adjustHours(noOfHours: -3)
         let sut = MealEventDetailsViewModel(mealEvent: commonMealEvent, dataController: dataController, imageDownloader: imageDownloader, mode: .edit)
-        XCTAssertEqual(sut.canEditMeal, false)
+        XCTAssertEqual(sut.canEditMeal, true)
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/MealEventDetailsViewModelTests.swift]: Why you change the value from false to true. Change it to false

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockLoginClient.swift [hunk 8]
```diff
@@ -26,7 +25,7 @@ class MockLoginClient: LoginCredentialsLoginClient {
         functionCalls.append("logout")
     }
 
-    func refreshLogin(completion: ((Result<AccessCredentials>) -> Void)?) {
+    func refreshLogin(completion: ((Result<AccessCredentials,Error>) -> Void)?) {
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockLoginClient.swift]: Replace value ((Result<AccessCredentials,Error>) -> Void) to accessCredentialsCompletion

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockLoginClient.swift [hunk 9]
```diff
@@ -6,7 +6,6 @@
 //  Copyright © 2017 Intrepid. All rights reserved.
 //
 
-import Intrepid
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockLoginClient.swift]: remove the space

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift [hunk 10]
```diff
@@ -8,27 +8,26 @@
...
 class MockPilotAPIClient: PilotAPIClient {
-    var mockCreateMealEventResponse: Result<RealmMealEvent> = .success(RealmMealEvent())
-    var mockCreateImageURLResponse: Result<String> = .success("")
-    var mockCreateLocationResponse: Result<Void> = .success(())
+    var mockCreateMealEventResponse: Result<RealmMealEvent,Error> = .success(RealmMealEvent())
+    var mockCreateImageURLResponse: Result<String,Error> = .success("")
+    var mockCreateLocationResponse: Result<Void,Error> = .success(())
 
-    override func createMeal(_ meal: Meal, completion: ((Result<RealmMeal>) -> Void)?) {
+    override func createMeal(_ meal: Meal, completion: ((Result<RealmMeal,Error>) -> Void)?) {
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift]: Change ((Result<RealmMeal,Error>) -> Void) to relamMealCompletion

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift [hunk 11]
```diff
@@ -8,27 +8,26 @@
...
         completion?(.success(RealmMeal()))
     }
 
-    override func createMealEvent(_ mealEvent: MealEvent, completion: ((Result<RealmMealEvent>) -> Void)?) {
+    override func createMealEvent(_ mealEvent: MealEvent, completion: ((Result<RealmMealEvent,Error>) -> Void)?) {
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift]: Change ((Result< RealmMealEvent,Error>) -> Void) to relamMealEventCompletion

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift [hunk 12]
```diff
@@ -8,27 +8,26 @@
...
         completion?(mockCreateMealEventResponse)
     }
 
-    override func createLocation(_ location: Location, completion: ((Result<Void>) -> Void)?) {
+    func createLocation(_ location: Location, completion: ((Result<Void,Error>) -> Void)?) {
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift]: Change ((Result<Void,Error>) -> Void) to voidRequestCompletion

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift [hunk 13]
```diff
@@ -23,8 +22,8 @@ class PilotLoginClientTests: XCTestCase {
     }()
 
     private var credentials: UserLoginCredentials {
-        let userName = "Participant99"
-        let password = "Password123"
+        let userName = "Participant41"
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift]: revert this change

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift [hunk 14]
```diff
@@ -35,8 +34,8 @@ class PilotLoginClientTests: XCTestCase {
                     "updated_at": "2017-03-02T21:48:35.579741Z",
                     "inserted_at": "2017-03-02T21:48:35.574107Z",
                     "id": "1dabcbbe-d2ce-4091-9d30-199d6ae833ad",
-                    "username": "Participant99",
-                    "device_id": "D9F0A23B-E5B0-4509-9F31-B37F2FB131C1"
+                    "username": "Participant41",
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift]: revert this change

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift [hunk 15]
```diff
@@ -71,15 +70,15 @@ class PilotLoginClientTests: XCTestCase {
 
         waitForExpectations(timeout: 1) { [weak self] error in
             XCTAssertNil(error)
-
+
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift]: Remove space here

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift [hunk 16]
```diff
@@ -244,36 +250,36 @@ fileprivate final class MockAccessCredentialsLoginClient: AccessCredentialsLogin
...
 
-    init(mockResult: Result<Data?>) {
+    init(mockResult: Result<Data?,Error>) {
         self.mockResult = mockResult
-        super.init(session: .shared, decoder: JSONDecoder.CallistoJSONDecoder())
+        super.init()
     }
 
-    fileprivate override func sendRequest(_ request: URLRequest, completion: ((Result<Data?>) -> Void)?) {
+    fileprivate func sendRequest(_ request: URLRequest, completion: ((Result<Data?,Error>) -> Void)?) {
```
Comment (rikky24Chavhan) [inline Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift]: change ((Result<Data?,Error>) -> Void) to defaultRequestCompletion

**Review by rikky24Chavhan — COMMENTED** [review]
Please refer below review comments

File: Mobl_ios_Callisto-master/MealTrackingPilot.xcodeproj/project.pbxproj (hunk 1 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilot.xcodeproj/project.pbxproj:1507]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/DemoPilotAPIClientTests.swift (hunk 2 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/DemoPilotAPIClientTests.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/DemoPilotAPIClientTests.swift (hunk 3 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/DemoPilotAPIClientTests.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/LocationTests.swift (hunk 5 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/LocationTests.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/LocationTests.swift (hunk 6 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/LocationTests.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/HealthKitTests.swift (hunk 4 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/HealthKitTests.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/MealEventDetailsViewModelTests.swift (hunk 7 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/MealEventDetailsViewModelTests.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockLoginClient.swift (hunk 8 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockLoginClient.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockLoginClient.swift (hunk 9 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockLoginClient.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift (hunk 11 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift (hunk 10 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift (hunk 12 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/Mocks/MockPilotAPIClient.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift (hunk 13 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift (hunk 14 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift (hunk 15 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift]: Done

File: Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift (hunk 16 above)
Comment (LTTS-sugapriya) [reply Mobl_ios_Callisto-master/MealTrackingPilotTests/PilotLoginClientTests.swift]: Done
